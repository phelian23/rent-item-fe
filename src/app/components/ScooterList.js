import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { displayScooter, getAllScooters } from '../redux/actions/items';

const ScooterList = () => {
  const dispatch = useDispatch();
  const scooters = useSelector((state) => state.items.scooters);

  useEffect(() => {
    dispatch(getAllScooters());
  }, [dispatch]);

  return (
    <div className="scoot-list-cont">
      <div className="text-center p-lg-3 header-det">
        <h1>Scooter List</h1>
      </div>
      <div className="list-wrapper">
        <ul className="d-flex list-con">
          {scooters.map((scooter) => (
            <li className="list-group-item col-md-4" key={scooter.id}>
              <Link
                to={`/scooter/${scooter.id}`}
                onClick={() => dispatch(displayScooter(scooter.id))}
                className="links"
              >
                <div className="scoot-card">
                  <img
                    className="card-img-top"
                    src={scooter.photo}
                    alt="Card cap"
                  />
                  <div className="card-body d-flex justify-content-between mt-4">
                    <h5 className="card-title">{scooter.name}</h5>
                    <p className="card-text">{scooter.price}</p>
                  </div>
                </div>
              </Link>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default ScooterList;
