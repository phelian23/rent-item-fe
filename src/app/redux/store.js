import { configureStore } from '@reduxjs/toolkit';
import { combineReducers, compose, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import logger from 'redux-logger';
import itemReducer from './reducers/itemReducer';
import userReducer from './reducers/userReducer';
import favouriteReducer from './reducers/favouriteReducer';

const reducer = combineReducers({
  items: itemReducer,
  user: userReducer,
  favourites: favouriteReducer,

  // Add other reducers here
});

const composed = compose(applyMiddleware(thunkMiddleware, logger));

const store = configureStore({ reducer, composed });

export default store;
