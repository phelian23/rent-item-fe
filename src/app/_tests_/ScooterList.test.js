import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import ScooterList from '../components/ScooterList';

describe('ScooterList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <ScooterList />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
