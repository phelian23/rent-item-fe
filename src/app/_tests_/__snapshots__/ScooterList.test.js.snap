// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScooterList renders correctly 1`] = `
<div
  className="scoot-list-cont"
>
  <div
    className="text-center p-lg-3 header-det"
  >
    <h1>
      Scooter List
    </h1>
  </div>
  <div
    className="list-wrapper"
  >
    <ul
      className="d-flex list-con"
    />
  </div>
</div>
`;
